tendermint/src/lite_impl/signed_header.rs

```diff
@@ -52,68 +48,89 @@ impl lite::Commit for block::signed_header::SignedHeader {
-                    if let Some(header_hash) = precommit.header_hash() {
-                        if header_hash != self.header_hash() {
-                            fail!(
-                                Kind::ImplementationSpecific,
-                                "validator({}) voted for header {}, but current header is {}",
-                                precommit.validator_address,
-                                header_hash,
-                                self.header_hash()
-                            );
-                        }
-                    }
-
-                    // returns FaultyFullNode error if it detects a signer isn't present in the validator set
-                    if vals.validator(precommit.validator_address) == None {
-                        let reason = format!(
-                            "Found a faulty signer ({}) not present in the validator set ({})",
-                            precommit.validator_address,
-                            vals.hash()
-                        );
-                        fail!(Kind::FaultyFullNode, reason);
-                    }
+        for commit_sig in self.commit.signatures.iter() {
+            // returns FaultyFullNode error if it detects a signer that is not present in the validator set
+            if let Some(val_addr) = commit_sig.validator_address {
+                if vals.validator(val_addr) == None {
+                    let reason = format!(
+                        "Found a faulty signer ({}) not present in the validator set ({})",
+                        val_addr,
+                        vals.hash()
+                    );
+                    fail!(Kind::FaultyFullNode, reason);
                 }
-                None => (),
             }
         }
 
         Ok(())
     }
 }
 
+fn commit_to_votes(commit: block::Commit) -> Result<Vec<vote::Vote>, Error> {
+    let mut votes: Vec<vote::Vote> = Default::default();
+    for (i, commit_sig) in commit.signatures.iter().enumerate() {
+        if commit_sig.is_absent() {
+            continue;
+        }
+
+        match commit_sig.validator_address {
+            Some(val_addr) => {
+                if let Some(sig) = commit_sig.signature.clone() {
+                    let vote = vote::Vote {
+                        vote_type: vote::Type::Precommit,
+                        height: commit.height,
+                        round: commit.round,
+                        block_id: Option::from(commit.block_id.clone()),
+                        timestamp: commit_sig.timestamp,
+                        validator_address: val_addr,
+                        validator_index: i as u64,
+                        signature: sig,
+                    };
+                    votes.push(vote);
+                }
+            }
+            None => {
+                fail!(
+                    Kind::ImplementationSpecific,
+                    "validator address missing in commit_sig {:#?}",
```

Comment:
I'm not familiar with the `#?` formatting directive. Couldn't find anything about it: https://doc.rust-lang.org/std/fmt/#formatting-traits
Did the `#` sneak in accidentally?